core/src/database/tx_sender.rs

```diff
@@ -2,83 +2,212 @@
 //!
 //! This module includes database functions which are mainly used by the transaction sender.
 
-use super::{
-    wrapper::{ScriptBufDB, TxidDB},
-    Database, DatabaseTransaction,
+use super::{wrapper::TxidDB, Database, DatabaseTransaction};
+use crate::{
+    errors::BridgeError,
+    execute_query_with_tx,
+    tx_sender::{ActivedWithOutpoint, ActivedWithTxid, FeePayingType},
 };
-use crate::{errors::BridgeError, execute_query_with_tx};
 use bitcoin::{
     consensus::{deserialize, serialize},
-    Amount, FeeRate, ScriptBuf, Transaction, Txid,
+    Amount, FeeRate, Transaction, Txid,
 };
-use std::str::FromStr;
+use std::ops::DerefMut; // Add this at the top with other imports
 
 impl Database {
-    /// Saves a fee payer transaction to the database.
-    ///
-    /// # Arguments
-    /// * `bumped_txid` - The txid of the bumped transaction
-    /// * `fee_payer_txid` - The txid of the fee payer transaction
-    /// * `vout` - The output index of the UTXO
-    /// * `script_pubkey` - The script pubkey of the UTXO
-    /// * `amount` - The amount in satoshis
-    pub async fn save_fee_payer_tx(
+    pub async fn confirm_transactions(
```

Comment:
this query looks really unoptimized and very complicated. Please write clear docs on what each query does (beyond the SQL operation like update/delete).

We should also look into understanding this operation's performance, optimizing this query, investigating required indices.